packages/@ember/-internals/routing/lib/system/router.ts

```diff
@@ -735,7 +735,12 @@ class EmberRouter extends EmberObject {
   _serializeQueryParam(value: unknown, type: string) {
     if (value === null || value === undefined) {
       return value;
-    } else if (type === 'array') {
+    } else if (type === 'array' && value instanceof Array) {
```

Comment:
I think we should use `Array.isArray` here (not `instanceof Array`)

Reply:
Done